R/function_options.R

```diff
@@ -0,0 +1,114 @@
+extendr_function_config <- rlang::env(
+  known_options = tibble::tribble(
+    ~Name, ~Ptype,
+    "use_try_from", logical(),
+    "r_name", character(),
+    "mod_name", character(),
+  )
+)
+
+#' Converts a list of user-specified options into a data frame containing `Name` and `RustValue`
+#'
+#' @param options A list of user-specified options.
+#' @param suppress_warnings Logical, suppresses warnigns if `TRUE`.
+#' @noRd
+convert_function_options <- function(options, suppress_warnings) {
+  if (rlang::is_null(options) || rlang::is_empty(options)) {
+    return(tibble::tibble(Name = character(), RustValue = character()))
+  }
+
+  if (!rlang::is_list(options) || !rlang::is_named(options)) {
+    cli::cli_abort("Extendr function options should be either a named {.code list()} or {.code NULL}.")
+  }
+
+  options_table <- tibble::tibble(Name = rlang::names2(options), Value = unname(options)) %>%
+    dplyr::left_join(extendr_function_config$known_options, by = "Name")
+
+  options_table <- options_table %>%
+    dplyr::rows_update(
```

Comment:
Yep, the problem here is that I cannot simply vectorize operations, because `vctrs::vec_cast` will fail on `NULL`, so I need to explicitly check 'if `NULL` do this otherwise cast'. I'll see if I can make it better.

Reply:
I found no better implementation right now. Let's keep it as is.

Reply:
Hard to tell. I can check if I can easily roll back these particular changes

Reply:
Done